api/schema/application_aws_resources_resource.go

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) Illumio, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package schema
+
+import (
+	resource_schema "github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var (
+	applicationAwsResourcesResource = Resource{
+		TypeName: "application_aws_resources",
+		Schema: resource_schema.Schema{
+			Version:     1,
+			Description: "Resources associated with an application in the Illumio CloudSecure platform.",
+			Attributes: map[string]resource_schema.Attribute{
+				IDFieldName: idAttribute,
+				"application_id": resource_schema.StringAttribute{
+					Description: "ID of the application.",
+					Required:    true,
+				},
+				"account_id": resource_schema.StringAttribute{
+					Description: "ID of the AWS account.",
+					Required:    true,
+				},
+				"aws_arns": resource_schema.SetAttribute{
+					ElementType: types.StringType,
+					Description: "ARNs of AWS resources to associate with the Cloudsecure application",
+					Optional:    true,
+				},
+				"aws_directconnect_connections": resource_schema.SetAttribute{
+					ElementType: types.StringType,
+					Description: "IDs of AWS direct connect connections to associate with Cloudsecure Application.",
+					Optional:    true,
+				},
+				"aws_directconnect_virtualinterfaces": resource_schema.SetAttribute{
+					ElementType: types.StringType,
+					Description: "IDs of AWS direct connect virtual interfaces to associate with Cloudsecure Application.",
+					Optional:    true,
+				},
+				"aws_ec2_customergateways": resource_schema.SetAttribute{
+					ElementType: types.StringType,
+					Description: "IDs of AWS ec2 customer gateways to associate with Cloudsecure Application.",
+					Optional:    true,
+				},
+				"aws_ec2_egressonlyinternetgateways": resource_schema.SetAttribute{
+					ElementType: types.StringType,
+					Description: "IDs of AWS ec2 egress only internet gateways to associate with Cloudsecure Application.",
+					Optional:    true,
+				},
+				"aws_ec2_eips": resource_schema.SetAttribute{
+					ElementType: types.StringType,
+					Description: "IDs of AWS ec2 eips to associate with Cloudsecure Application.",
+					Optional:    true,
+				},
+				"aws_ec2_flowlogs": resource_schema.SetAttribute{
+					ElementType: types.StringType,
+					Description: "IDs of AWS ec2 flow logs to associate with Cloudsecure Application.",
+					Optional:    true,
+				},
+				"aws_ec2_instances": resource_schema.SetAttribute{
+					ElementType: types.StringType,
+					Description: "IDs of AWS ec2 instances to associate with Cloudsecure Application.",
+					Optional:    true,
+				},
+				"aws_ec2_instanceconnectendpoints": resource_schema.SetAttribute{
+					ElementType: types.StringType,
+					Description: "IDs of AWS ec2 instance connect endpoints to associate with Cloudsecure Application.",
+					Optional:    true,
+				},
+				"aws_ec2_internetgateways": resource_schema.SetAttribute{
```

Comment:
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/internet_gateway
```suggestion
				"aws_internet_gateway_ids": resource_schema.ListAttribute{
```